### DISCUSSION TITLE
Isn't 2160 blocks 12h instead of 36h to guarantee blocks be immutable?

### DISCUSSION THREAD

**Discussion by eyeinsky** [CardanoSolutions/kupo#136] [Q&A] [answered]
Hi, noob question about Cardano slotting: since one block is 20 seconds then how come the safe time threshold is cited as 36h -- isn't 2160 blocks * 20 seconds = 43200 seconds = 12 hours?

**Comment by KtorZ** [comment, marked as answer]
It's a little bit more complicated. One block isn't created every 20s. On average, there's a block every 20s if the chain only contains honest actors. So in principle you cannot really make assumptions about the time it takes to see N blocks. Could be N*20s, could be slightly less or it could be more. 

Yet, in Ouroboros Praos, the protocol gives you another guarantee: at least k honest blocks are produced in a window of 3k/f slots; where k is the security parameter (k=2160 on mainnet) and f the active slot coefficient (f=0.05 on mainnet). 

Thus on Mainnet, This tells you that in a window of 129600 slots, the protocol guarantees that you see *at least* k blocks. With a slot length of 1s, that's 36h.

**Reply by eyeinsky** [reply]
Thank you, that makes it clear! :)
